rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -294,6 +382,23 @@ ImageData ROSImageTexture::setFormatAndNormalizeDataIfNecessary(ImageData image_
       image_data.size_);
     image_data.pixel_format_ = Ogre::PF_BYTE_L;
     image_data.data_ptr_ = &buffer[0];
+  } else if ( // NOLINT enforces bracket on the same line, which makes code unreadable
+    image_data.encoding_ == sensor_msgs::image_encodings::YUV422 ||
+    image_data.encoding_ == sensor_msgs::image_encodings::YUV422_YUY2)
+  {
+    int new_size = image_data.size_ * 3 / 2;
+    if (!bufferptr_) {
+      bufferptr_ = std::make_shared<std::vector<uint8_t>>(new_size);
+    } else if (static_cast<int>(bufferptr_->size()) != new_size) {
+      bufferptr_->resize(new_size, 0);
+    }
+
+    imageConvertYUV422ToRGB(
+      bufferptr_->data(), const_cast<uint8_t *>(image_data.data_ptr_),
+      0, height_, width_, stride_, image_data.encoding_);
+    image_data.pixel_format_ = Ogre::PF_BYTE_RGB;
+    image_data.data_ptr_ = bufferptr_->data();
```

Comment:
This is a good solution to the problem of the lifetime of the data_ptr.  We probably need to apply this solution up above for `TYPE_32FC1`, `TYPE_16UC1`, `TYPE_16S1`, and `TYPE_MONO` as well, but that can be done in a separate PR.  However, I would suggest you open an issue to track that.

Reply:
Issue #706 made to track this problem.